scripts/process-trace.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+
+"""Mbed TLS instrumentation post-processor
+
+Compile with -DMBEDTLS_INST and -DMBEDTLS_DEBUG_C. Capture the output to
+a log file and pass it to this post processor. It is designed to generate
+a table of which primitives are used in which handshake stages, and how
+many bytes are used per context (if applicable).
+"""
+
+## Copyright The Mbed TLS Contributors
+## SPDX-License-Identifier: Apache-2.0
+##
+## Licensed under the Apache License, Version 2.0 (the "License"); you may
+## not use this file except in compliance with the License.
+## You may obtain a copy of the License at
+##
+## http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+## WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+import sys
+import re
+import json
+
+
+# We use this global to ignore all SHA operations when inside ECDSA
+# mbedtls uses SHA in our config, so it is a hack to remove misleading
+# SHA operations that are implementation-specific.
+FENCE_shablock = False
+
+current_state = -1
```

Comment:
Please don't use global state. Put everything that has state in a class and instantiate that class in `main`.